### PR TITLE
Track events for the dashboard top earning pages widget.

### PR DESCRIPTION
**Pull request by benbowler**
## Summary

Addresses issue:

- #8212

## Relevant technical choices

 - In addition to the IB, I had to update `assets/js/googlesitekit/widgets/util/get-widget-component-props.js` to forward the ref through to the underlaying `Widget` component:

https://github.com/google/site-kit-wp/blob/fe7905c6c3125f2aa82a03a81fcafd7f31ed3a54/assets/js/googlesitekit/widgets/util/get-widget-component-props.js#L65-L73

 - In testing I found [this bug](https://github.com/google/site-kit-wp/issues/8212#issuecomment-1954275748) which I addressed by using state to re-render the `DashboardTopEarningPagesWidgetGA4` component when the underlaying ref dom element was correctly set:

https://github.com/google/site-kit-wp/blob/034818eb3c27089d23bcf6a17eaa1b0568de8765/assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.js#L125-L135

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.js [hunk 1]
```diff
@@ -117,6 +120,55 @@ function DashboardTopEarningPagesWidgetGA4( {
...
+	// This meant that if you scrolled slowly, the trackingRef would be null when the intersection observer
+	// was created, and the observer would never detect the component as in view.
+	// Full discussion: https://github.com/google/site-kit-wp/issues/8212#issuecomment-1954275748
+	const [ trackingRefReady, setTrackingRefReady ] = useState( false );
+	const updateTrackingRef = ( el ) => {
+		trackingRef.current = el;
+		if ( el && ! trackingRefReady ) {
+			setTrackingRefReady( true );
+		}
+	};
```
Comment (tofumatt) [inline assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.js:135]: Let's avoid shorthand variables like `el` in favour of `element`, when possible:
```suggestion
	const updateTrackingRef = ( element ) => {
		trackingRef.current = element;
		if ( element && ! trackingRefReady ) {
			setTrackingRefReady( true );
		}
	};
```

**Review by tofumatt — APPROVED** [review]
Looks good, thanks for finding the fix and all the context around it. 👍🏻 

Just one variable name tweak and then I'll merge this 👍🏻

**Review by tofumatt — CHANGES_REQUESTED** [review]
Oh, actually, just one thing here.

Given #8049 isn't merged yet, can you provide a way for QA to test the linked state of the component to test the `trackEvent` calls there? I assume you used some JS console code to trigger the linked state—can you add that in so that part can be QA'd?

**Comment by benbowler** [comment]
@tofumatt added a section to the QAB on how to test the linked state.

**Review by tofumatt — APPROVED** [review]
Awesome, thanks @benbowler! 👍🏻

**Pull request merged**